### PR TITLE
IR-714 add sequences for initial mapping work from syscon to generate from our APIs

### PR DESCRIPTION
**Pull request by Mjwillis**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/dto/nomis/NomisOffenderParty.kt [hunk 1]
```diff
@@ -8,6 +8,8 @@ import java.time.LocalDateTime
 data class NomisOffenderParty(
   @Schema(description = "Offender involved in the incident")
   val offender: NomisOffender,
+  @Schema(description = "The sequence number providing an order for a list of offender parties")
+  val sequence: Int? = null,
```
Comment (Mjwillis) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/dto/nomis/NomisOffenderParty.kt:12]: null will be removed once sync provides them

File: src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/dto/nomis/NomisRequirement.kt [hunk 2]
```diff
@@ -7,6 +7,8 @@ import java.time.LocalDateTime
 
 @JsonInclude(JsonInclude.Include.NON_NULL)
 data class NomisRequirement(
+  @Schema(description = "The sequence number providing an order for a list of requirements")
+  val sequence: Int? = null,
```
Comment (Mjwillis) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/dto/nomis/NomisRequirement.kt:11]: null will be removed once sync provides them

File: src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/dto/nomis/NomisStaffParty.kt [hunk 3]
```diff
@@ -8,6 +8,8 @@ import java.time.LocalDateTime
 data class NomisStaffParty(
   @Schema(description = "Staff involved in the incident")
   val staff: NomisStaff,
+  @Schema(description = "The sequence number providing an order for a list of staff parties")
+  val sequence: Int? = null,
```
Comment (Mjwillis) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/dto/nomis/NomisStaffParty.kt:12]: null will be removed once sync provides them

**Review by karenmillermoj — APPROVED** [review]
lgtm

**Pull request merged**